platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateFileManager.java

```diff
@@ -216,6 +216,117 @@ public int getTaskQueueSize() {
         return taskQueue.size();
     }
 
+    /**
+     * Method to be called when a state has been successfully written to disk.
+     * <p>
+     * This method shouldn't be called if the state was written out of band.
+     *
+     * @param reservedState the state that was written to disk
+     * @param directory     the directory where the state was written
+     * @param start         the nano start time of the state writing process
+     */
+    private void stateWrittenToDisk(
+            @NonNull final SignedState reservedState, @NonNull final Path directory, final long start) {
+
+        final long round = reservedState.getRound();
+        if (round > latestSavedStateRound.get()) {
+            latestSavedStateRound.set(round);
+        }
+
+        metrics.getWriteStateToDiskTimeMetric().update(TimeUnit.NANOSECONDS.toMillis(time.nanoTime() - start));
+        statusActionSubmitter.submitStatusAction(new StateWrittenToDiskAction(round));
+        stateToDiskAttemptConsumer.stateToDiskAttempt(reservedState, directory, true);
+
+        reservedState.stateSavedToDisk();
+    }
+
+    /**
+     * Method to be called when a state is being written to disk in-band, but it lacks signatures.
+     * <p>
+     * This method shouldn't be called if the state was written out of band.
+     *
+     * @param reservedState the state being written to disk
+     */
+    private void stateLacksSignatures(@NonNull final SignedState reservedState) {
+        metrics.getTotalUnsignedDiskStatesMetric().increment();
+        final long newCount = metrics.getTotalUnsignedDiskStatesMetric().get();
+
+        logger.error(
+                EXCEPTION.getMarker(),
+                new InsufficientSignaturesPayload(("State written to disk for round %d did not have enough signatures. "
+                                + "Collected signatures representing %d/%d weight. "
+                                + "Total unsigned disk states so far: %d.")
+                        .formatted(
+                                reservedState.getRound(),
+                                reservedState.getSigningWeight(),
+                                reservedState.getAddressBook().getTotalWeight(),
+                                newCount)));
+    }
+
+    /**
+     * Create an element that can be offered to the {@link #taskQueue}
+     *
+     * @param reservedSignedState  the reserved signed state to be written to disk
+     * @param directory            the directory where the signed state will be written
+     * @param reason               the reason this state is being written to disk
+     * @param finishedCallback     a function that is called after state writing is complete
+     * @param outOfBand            whether this state has been requested to be written out of band
+     * @param stateLacksSignatures whether the state being written lacks signatures
+     */
+    private void createTaskQueueElement(
```

Comment:
This method does not create a task queue element, hence the void return. The method itself _is_ the element, wrapped in a lambda. I suggest `saveStateTask(...)`